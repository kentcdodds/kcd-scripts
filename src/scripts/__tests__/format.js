import cases from 'jest-in-case'
import {unquoteSerializer, winPathSerializer} from './helpers/serializers'

expect.addSnapshotSerializer(unquoteSerializer)
expect.addSnapshotSerializer(winPathSerializer)

cases(
  'format',
  ({args}) => {
    // beforeEach
    const {sync: crossSpawnSyncMock} = require('cross-spawn')
    const originalExit = process.exit
    const originalArgv = process.argv
    const utils = require('../../utils')
    utils.resolveBin = (modName, {executable = modName} = {}) => executable
    process.exit = jest.fn()

    // tests
    process.argv = ['node', '../format', ...args]
    crossSpawnSyncMock.mockClear()
    require('../format')
    expect(crossSpawnSyncMock).toHaveBeenCalledTimes(1)
    const [firstCall] = crossSpawnSyncMock.mock.calls
    const [script, calledArgs] = firstCall
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot()

    // afterEach
    process.exit = originalExit
    process.argv = originalArgv
    jest.resetModules()
  },
  {
    'calls prettier CLI with args': {
      args: ['my-src/**/*.js'],
    },
    '--no-write prevents --write argument from being added': {
      args: ['--no-write'],
    },
    '--config arg can be used for a custom config': {
      args: ['--config', './my-config.js'],
    },
    '--ignore-path arg can be used for a custom ignore file': {
      args: ['--ignore-path', './.myignore'],
    },
  },
)
