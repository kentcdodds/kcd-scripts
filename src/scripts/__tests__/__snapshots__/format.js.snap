// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format --config arg can be used for a custom config 1`] = `prettier --ignore-path ./src/config/prettierignore --write **/*.+(js|json|less|css|ts|tsx|md) --config ./my-config.js`;

exports[`format --ignore-path arg can be used for a custom ignore file 1`] = `prettier --config ./src/config/prettierrc.js --write **/*.+(js|json|less|css|ts|tsx|md) --ignore-path ./.myignore`;

exports[`format --no-write prevents --write argument from being added 1`] = `prettier --config ./src/config/prettierrc.js --ignore-path ./src/config/prettierignore **/*.+(js|json|less|css|ts|tsx|md) --no-write`;

exports[`format calls prettier CLI with args 1`] = `prettier --config ./src/config/prettierrc.js --ignore-path ./src/config/prettierignore --write my-src/**/*.js`;
